src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-wizard/rgw-multisite-wizard.component.ts

```diff
@@ -0,0 +1,256 @@
+import { Component, OnInit } from '@angular/core';
+import { UntypedFormControl, Validators } from '@angular/forms';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import { Subscription, forkJoin } from 'rxjs';
+import { ActionLabelsI18n } from '~/app/shared/constants/app.constants';
+import { CdFormGroup } from '~/app/shared/forms/cd-form-group';
+import { WizardStepModel } from '~/app/shared/models/wizard-steps';
+import { WizardStepsService } from '~/app/shared/services/wizard-steps.service';
+import { RgwDaemonService } from '~/app/shared/api/rgw-daemon.service';
+import { RgwDaemon } from '../models/rgw-daemon';
+import { MultiClusterService } from '~/app/shared/api/multi-cluster.service';
+import { RgwMultisiteService } from '~/app/shared/api/rgw-multisite.service';
+import { Icons } from '~/app/shared/enum/icons.enum';
+import { SelectOption } from '~/app/shared/components/select/select-option.model';
+import _ from 'lodash';
+import { SelectMessages } from '~/app/shared/components/select/select-messages.model';
+import { NotificationType } from '~/app/shared/enum/notification-type.enum';
+import { NotificationService } from '~/app/shared/services/notification.service';
+import { Router } from '@angular/router';
+import { map, switchMap } from 'rxjs/operators';
+
+@Component({
+  selector: 'cd-rgw-multisite-wizard',
+  templateUrl: './rgw-multisite-wizard.component.html',
+  styleUrls: ['./rgw-multisite-wizard.component.scss']
+})
+export class RgwMultisiteWizardComponent implements OnInit {
+  multisiteSetupForm: CdFormGroup;
+  currentStep: WizardStepModel;
+  currentStepSub: Subscription;
+  permissions: Permissions;
+  stepTitles = ['Create Realm & Zonegroup', 'Create Zone', 'Select Cluster'];
+  stepsToSkip: { [steps: string]: boolean } = {};
+  daemons: RgwDaemon[] = [];
+  selectedCluster = '';
+  clusterDetailsArray: any;
+  isMultiCluster = false;
+  exportTokenForm: CdFormGroup;
+  realms: any;
+  loading = false;
+  icons = Icons;
+  rgwEndpoints: { value: any[]; options: any[]; messages: any };
+
+  constructor(
+    private wizardStepsService: WizardStepsService,
+    public activeModal: NgbActiveModal,
+    public actionLabels: ActionLabelsI18n,
+    private rgwDaemonService: RgwDaemonService,
+    private multiClusterService: MultiClusterService,
+    private rgwMultisiteService: RgwMultisiteService,
+    public notificationService: NotificationService,
+    private router: Router
+  ) {
+    this.currentStepSub = this.wizardStepsService
+      .getCurrentStep()
+      .subscribe((step: WizardStepModel) => {
+        this.currentStep = step;
+      });
+    this.currentStep.stepIndex = 1;
+    this.createForm();
+    this.rgwEndpoints = {
+      value: [],
+      options: [],
+      messages: new SelectMessages({
+        empty: $localize`There are no endpoints.`,
+        filter: $localize`Add endpoints`
```

Comment:
if the intention is to select from the provided list then
```suggestion
        filter: $localize`Select endpoints`
```